delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/organizations/OrganizationsImpl.scala

```diff
@@ -1,219 +1,127 @@
       _        <- IO.parTraverseUnordered(scopeInitializations)(_.onOrganizationCreation(resource.value, caller))
                     .void
                     .mapError(OrganizationInitializationFailed)
-                    .named("initializeOrganization", moduleType)
+                    .span("initializeOrganization")
     } yield resource
 
   override def update(
       label: Label,
       description: Option[String],
-      rev: Long
+      rev: Int
   )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
-    eval(UpdateOrganization(label, rev, description, caller)).named("updateOrganization", moduleType)
+    eval(UpdateOrganization(label, rev, description, caller)).span("updateOrganization")
 
   override def deprecate(
       label: Label,
-      rev: Long
+      rev: Int
   )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
-    eval(DeprecateOrganization(label, rev, caller)).named("deprecateOrganization", moduleType)
+    eval(DeprecateOrganization(label, rev, caller)).span("deprecateOrganization")
 
   override def fetch(label: Label): IO[OrganizationNotFound, OrganizationResource] =
-    agg
-      .state(label.value)
-      .map(_.toResource)
-      .flatMap(IO.fromOption(_, OrganizationNotFound(label)))
-      .named("fetchOrganization", moduleType)
+    log.state(label, OrganizationNotFound(label)).map(_.toResource).span("fetchOrganization")
 
-  override def fetchAt(label: Label, rev: Long): IO[OrganizationRejection.NotFound, OrganizationResource] =
-    eventLog
-      .fetchStateAt(persistenceId(moduleType, label.value), rev, Initial, Organizations.next)
-      .bimap(RevisionNotFound(rev, _), _.toResource)
-      .flatMap(IO.fromOption(_, OrganizationNotFound(label)))
-      .named("fetchOrganizationAt", moduleType)
+  override def fetchAt(label: Label, rev: Int): IO[OrganizationRejection.NotFound, OrganizationResource] = {
+    log
+      .state(label, rev, OrganizationNotFound(label), RevisionNotFound)
+      .map(_.toResource)
+      .span("fetchOrganizationAt")
+  }
 
   override def fetch(uuid: UUID): IO[OrganizationNotFound, OrganizationResource] =
-    fetchFromCache(uuid).flatMap(fetch).named("fetchOrganizationByUuid", moduleType)
-
-  override def fetchAt(uuid: UUID, rev: Long): IO[OrganizationRejection.NotFound, OrganizationResource] =
-    super.fetchAt(uuid, rev).named("fetchOrganizationAtByUuid", moduleType)
-
-  private def fetchFromCache(uuid: UUID): IO[OrganizationNotFound, Label]                     =
-    cache.collectFirstOr { case (label, resource) if resource.value.uuid == uuid => label }(OrganizationNotFound(uuid))
+    fetchFromCache(uuid).flatMap(fetch).span("fetchOrganizationByUuid")
+
+  override def fetchAt(uuid: UUID, rev: Int): IO[OrganizationRejection.NotFound, OrganizationResource] =
+    super.fetchAt(uuid, rev).span("fetchOrganizationAtByUuid")
+
+  private def fetchFromCache(uuid: UUID): IO[OrganizationNotFound, Label] = {
```

Comment:
I'm wondering if it's necessary to retain this cache instead of running the listing every time (with an access by uuid deprecation notice).